### PR TITLE
Jsx3 andrii

### PR DESCRIPTION
**Pull request by andrii-antiukhov**
In this files i made changes, but not all was refactored to JSX3 (need to do this)

```
client/shared/Fi.re
client/shared/React.re
client/shared/ReactOld.re
```

All components in folder `client/src/components/` was refactored to JSX3. But i did not run compiler, so maybe there are some errors.
@jchavarri Not sure about changing `React.re` to `ReactOld.re` I did this because last time me and Feihong found bug, so i created other `React.re` But i do not know do we need this for JSX3 Sorry. If not need thiod changes, i will remove `ReactOld` from all files in this PR.

### CODE REVIEW

File: client/shared/ButterToast.re
Comment (jchavarri) [inline client/shared/ButterToast.re:72]: Using `ReactOld` temporarily is ok while you work on this, but ultimately it's not needed and `ReactOld` should be removed before we merge the jsx3 migration.

File: client/shared/Fi.re
Comment (jchavarri) [inline client/shared/Fi.re:2]: This was not used?

File: client/src/Router.re
Comment (jchavarri) [inline client/src/Router.re:54]: I don't understand this change :/ why not using `React.null`? One less thing to change later on.

File: client/src/Router.rei
Comment (jchavarri) [inline client/src/Router.rei:27]: We should update these signatures to jsx3 as well, e.g.

```reason
[@react.component]
let make: unit => React.element;
```

File: client/src/Router.rei
Comment (jchavarri) [inline client/src/Router.rei:45]: We should update this to jsx3 as well:
```reason
[@react.component]
  let make:
    (
       ~route: Route.t,
      ~id: string=?,
      ~title: string=?,
      ~className: string=?,
      ~popup: bool=?,
      ~role: string=?,
      ~children: React.element,
      unit
    ) =>
    React.element;
```

File: client/src/Router.rei
Comment (jchavarri) [inline client/src/Router.rei:51]: This should be just `React.element`

File: client/src/components/Notify.re
Comment (jchavarri) [inline client/src/components/Notify.re:1]: Please undo all the changes from `require` to `import`. They will break the app. We want `require`.

File: client/src/components/Notify.re
Comment (jchavarri) [inline client/src/components/Notify.re:5]: `message` should be a labelled arg. Probably it was meant to be `children` 🤔

File: client/src/components/UI_DateTime.re
Comment (jchavarri) [inline client/src/components/UI_DateTime.re:26]: We can use `React.null` now.

File: client/src/components/UI_LoadingWrapper.re
Comment (jchavarri) [inline client/src/components/UI_LoadingWrapper.re:33]: If you look carefully at the previous code, this call to `clearTimeout` was done when unmounting. To keep the same behavior, it should be moved to the returned value of the effect, e.g.

```
 React.useEffect(() => {
    Some(() => clearTimeout(timeoutId^));
}
```

File: client/src/components/UI_LoadingWrapper.re
Comment (jchavarri) [inline client/src/components/UI_LoadingWrapper.re:65]: This is not the same behavior as before. It should be something like `children(displayLoading)`.

File: client/src/components/UI_UserInfo.re
Comment (jchavarri) [inline client/src/components/UI_UserInfo.re:8]: I don't think spread operator `...` is needed with jsx3. I am pretty sure this won't compile.

**Review by jchavarri — COMMENTED** [review]
@andrii-antiukhov this is looking good. please take a look at comment, I will merge after they are tackled, thanks!

File: client/shared/ButterToast.re
Comment (andrii-antiukhov) [reply client/shared/ButterToast.re]: Removed `ReacOld`

File: client/shared/Fi.re
Comment (andrii-antiukhov) [reply client/shared/Fi.re:2]: Looks like this icon does not used in code.

File: client/src/Router.re
Comment (andrii-antiukhov) [reply client/src/Router.re]: Remove `ReactOld`

File: client/src/components/Notify.re
Comment (andrii-antiukhov) [reply client/src/components/Notify.re]: Returned back.

File: client/src/Router.rei
Comment (andrii-antiukhov) [reply client/src/Router.rei]: Changed.

File: client/src/Router.rei
Comment (andrii-antiukhov) [reply client/src/Router.rei]: Changed.

File: client/src/Router.rei
Comment (andrii-antiukhov) [reply client/src/Router.rei]: Changed.

File: client/src/components/Notify.re
Comment (andrii-antiukhov) [reply client/src/components/Notify.re]: I changed it like labeled. But according code this variable must be string type, not children. Maybe.

File: client/src/components/UI_DateTime.re
Comment (andrii-antiukhov) [reply client/src/components/UI_DateTime.re]: Replaced `ReasonReact.null` to `React.null`

File: client/src/components/UI_LoadingWrapper.re
Comment (andrii-antiukhov) [reply client/src/components/UI_LoadingWrapper.re]: Thanks! Changed to return some value.

File: client/src/components/UI_UserInfo.re
Comment (andrii-antiukhov) [reply client/src/components/UI_UserInfo.re]: Removed

File: client/src/components/UI_LoadingWrapper.re
Comment (andrii-antiukhov) [reply client/src/components/UI_LoadingWrapper.re]: Sorry, i am not sure `children` can be function. But maybe. Thanks!

**Review by jchavarri — APPROVED** [review]
Looks good! Please take a look at these minor things and I think we can merge it ,thanks!

File: client/src/components/UI_DateTime.re
Comment (maxlibin) [reply client/src/components/UI_DateTime.re:23]: This component is already updated in jsx3 branch: https://github.com/Sketch-sh/sketch-sh/pull/321
Screenshot 2022-05-06 at 1 51 57 PM

Comment (andrii-antiukhov) [reply client/src/components/UI_DateTime.re:23]: @jchavarri @maxlibin Here i added this `[@react.component]` I think it need this code. If no need i will remove.

File: client/src/components/UI_SketchOwnerInfo.re
Comment (andrii-antiukhov) [reply client/src/components/UI_SketchOwnerInfo.re]: Changed.

File: client/src/components/UI_FullpageLoading.re
Comment (andrii-antiukhov) [reply client/src/components/UI_FullpageLoading.re]: Changed.

**Comment by andrii-antiukhov** [comment]
Hi @jchavarri Today i successfully up sketch on my local Mac and i did some refactoring. So not so bad was today. I refactored `Embed.re` and `ReLoadable.re` and fixed some other issues. Please, review this PR and you can merge it to branch `jsx3` if need.
For me and @maxlibin left to refactor 2 files
```
 src_editor/Editor_Blocks.re
 src_editor/Editor_Links.re
```
But maybe we will do this next time, because this files not so simple to refactor. Thanks!

**Comment by andrii-antiukhov** [comment]
@jchavarri Sorry i can not finish properly file `src_editor/Editor_Blocks.re` too complicated for me. I done main part but i do not know how to refactor some old code. So you can look and maybe can fix rest part of code. Sorry.
File `src_editor/Editor_Links.re` was refactored full. Sorry, i do not have types because can not up compiler, difficult to understand logic. So i hope i help you how i can. I tried do everything in best way, but not enough knowledge.
So me and Max waiting for your instructions for the next opensource day. If it is still make sense for us. Thanks!

**Pull request closed without merging**
